### DISCUSSION TITLE
NodeJS - Send attachments

### DISCUSSION THREAD

**Discussion by bernapastorini** [chatwoot/chatwoot#5149] [Chatwoot APIs] [answered]
Hello! I need help with something... I'm trying to send files through the ChatWoot API but I'm not able to do it...

I'm using NestJS, a NodeJS framework.

I make the requests using "HttpService" from "@nestjs/axios" package.

I am doing this:
![code](https://user-images.githubusercontent.com/56703142/181786823-0915c43d-35b1-485a-8af3-98be98c647cf.png)

The "file" variable is from type "fs.ReadStream", pointing to file location.

And I get this response:
{
  "id": 284,
  "content": null,
  "inbox_id": 2,
  "conversation_id": 54,
  "message_type": 1,
  "content_type": "text",
  "content_attributes": {},
  "created_at": 1659106073,
  "private": false,
  "source_id": null,
  "sender": {
    "id": 1,
    "name": "John",
    "available_name": "John",
    "avatar_url": "https://www.gravatar.com/avatar/0d722ac7bc3b3c92c030d0da9690d981?d=404",
    "type": "user",
    "availability_status": "online",
    "thumbnail": "https://www.gravatar.com/avatar/0d722ac7bc3b3c92c030d0da9690d981?d=404"
  }
}

What am I doing wrong?

I checked this post #1809  on the same topic, but I couldn't figure it out...

Thank you very much!!!

**Comment by bernapastorini** [comment]
Can anyone give me a hand with this? Thanks!

**Comment by bernapastorini** [comment, marked as answer]
In case it helps anyone, here is the solution to the problem: 

`const headers = {
      api_access_token: this.token,
      ...form.getHeaders(),
    }`

Adding **...form.getHeaders()** to the request headers, i solved the problem.

**Reply by joeenet** [reply]
Do you mind pasting your entire function, I have the same issue.

**Reply by joeenet** [reply]
I figure it out. Never mind

**Reply by bernapastorini** [reply]
Hello, the function is the same as the one I published.

Just change line 4 to
`form.append('attachments[]', file, fileName);`

and the headers by 

`headers: { api_access_token: this.token, ...form.getHeaders(), }`

Greetings!
